frontend/ui/widget/opdsbrowser.lua

```diff
@@ -645,7 +660,55 @@ function OPDSBrowser:showDownloads(item)
     UIManager:show(self.download_dialog)
 end
 
+function OPDSBrowser:browse(browse_url, username, password)
+  logger.info("launch url", browse_url)
+  table.insert(self.paths, {
+                 url = browse_url,
+                 username = username,
+                 password = password,
+  })
+  if not self:updateCatalog(browse_url, username, password) then
+    table.remove(self.paths)
+  end
+end
+
+function OPDSBrowser:browseSearchable(browse_url, username, password)
+    self.search_server_dialog = InputDialog:new{
+        title = _("Search OPDS catalog"),
+        input = "",
+        hint = _("Search string"),
+        input_hint = _("author:dumas alexandre"),
+        input_type = "string",
+        description = _("%s in url will be replaced by your input"),
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    callback = function()
+                        UIManager:close(self.search_server_dialog)
+                    end,
+                },
+                {
+                    text = _("Search"),
+                    is_enter_default = true,
+                    callback = function()
+                                UIManager:close(self.search_server_dialog)
+                                local search, _ = self.search_server_dialog:getInputText():gsub(" ", "+")
```

Comment:
If you want to discard the second one you can just use `local search = function that returns multiple things`.